lib/Doctrine/DBAL/DBALException.php

```diff
@@ -77,10 +77,22 @@ public static function invalidPdoInstance()
     }
 
     /**
+     * @param bool|null $url The URL that was provided in the connection parameters (if any).
+     *
      * @return \Doctrine\DBAL\DBALException
      */
-    public static function driverRequired()
+    public static function driverRequired($url = null)
     {
+        if ($url) {
+            return new self(
+                sprintf(
+                    "The options 'driver' or 'driverClass' are mandatory if a connection URL without scheme " .
+                    "is given to DriverManager::getConnection(). Given URL: %s",
+                    $url
+                )
+            );
+        }
```

Comment:
Errata: seen now that `'url'` may be passed in from multiple contexts. It is indeed better to let the exception class decide here.